controllers/bpfman-agent/common.go

```diff
@@ -633,6 +633,31 @@ func (r *ReconcilerCommon) handleProgDelete(
 	return internal.Unchanged, nil
 }
 
```

Comment:
// unloadAndDeletePrograms unloads and deletes BbpPrograms when the owning 
// *Program or BpfApplication is not being deleted itself, but something 
// has change such that the BpfPrograms are no longer needed.
```suggestion

```